lamindb_setup/_check_setup.py

```diff
@@ -1,33 +1,39 @@
 from lamin_utils import logger
+from typing import Optional
 import os
 from ._init_instance import reload_lamindb, reload_schema_modules
 from ._silence_loggers import silence_loggers
 from .core._settings_store import current_instance_settings_file
 
+_LAMINDB_CONNECTED_TO: Optional[str] = None
+
 _INSTANCE_NOT_SETUP_WARNING = """\
-You haven't yet setup an instance: Please call `ln.setup.init()` or `ln.setup.connect()`
+To use lamindb, you need to connect to an instance.
+
+Connect to an instance: `ln.connect()`. Init an instance: `ln.setup.init()`.
 """
 
 
-def check_instance_setup(from_lamindb: bool = False):
-    # silence_loggers is called twice within the if conditions for latency
-    # reasons
+def check_instance_setup(from_lamindb: bool = False) -> bool:
+    from .core.django import IS_SETUP, setup_django
+
+    if _LAMINDB_CONNECTED_TO is not None:
+        return True
+    if IS_SETUP:
```

Comment:
@Koncopd - this line makes it more rigorous because this measures whether django is setup or not.

Reply:
This is why now, we need the `_test` flag. I will try this.

(Just confirming what you said on the call before!)

Reply:
Tried it and it doesn't work because it doesn't attempt to access the SQLite file when run with `_test`. Let's see whether I can change this.

Reply:
No, this should not matter, lock is called in `_load_db`, which is called always on load / connect.